### PR TITLE
Compile tests as generator.

### PR DESCRIPTION
**Pull request by godofredoc**
## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by chinmaygarde** [comment]
Looks like the failure is related.

**Comment by godofredoc** [comment]
> Looks like the failure is related.
It required https://github.com/flutter/flutter/commit/be97d03a25851f70a8dc4d00b30f3dd4e445614c

File: ci/builders/linux_web_engine.json [hunk 1]
```diff
@@ -41,28 +41,50 @@
...
+        {
+            "archives": [],
+            "drone_dimensions": [
+                "device_type=none",
+                "os=Linux"
+            ],
+            "gclient_variables": {
+                "download_android_deps": false
+            },
+            "gn": [],
```
Comment (eyebrowsoffire) [inline ci/builders/linux_web_engine.json:74]: We want to just omit this field completely, right? Skip gn and ninja entirely.

File: ci/builders/linux_web_engine.json [hunk 2]
```diff
@@ -41,28 +41,50 @@
...
+            "name": "web_tests",
+            "ninja": {},
+            "generators": {
+                "pub_dirs": [
```
Comment (eyebrowsoffire) [inline ci/builders/linux_web_engine.json:78]: I don't think we need these. `felt` will automatically run pub on anything it needs

**Review by eyebrowsoffire — CHANGES_REQUESTED** [review]

File: ci/builders/linux_web_engine.json (hunk 1 above)
Comment (godofredoc) [reply ci/builders/linux_web_engine.json:74]: Not sure it will work but we can give it a try.

File: ci/builders/linux_web_engine.json (hunk 2 above)
Comment (godofredoc) [reply ci/builders/linux_web_engine.json:78]: ack

File: ci/builders/linux_web_engine.json (hunk 1 above)
Comment (godofredoc) [reply ci/builders/linux_web_engine.json]: ok, confirmed it worked!

**Review by eyebrowsoffire — APPROVED** [review]
LGTM!

**Pull request merged**